File: angular/src/app/model/RessourceHumaineHeritage/stagiaire.ts
import {RessourceHumaine} from '../ressource-humaine';
import {Ordinateur} from '../RessourceMaterielle/ordinateur';
import {Cursus} from '../cursus';

export class Stagiaire extends RessourceHumaine {
  constructor(private _profil?: string, private _ordinateur?: Ordinateur, private _cursus?: Cursus) {
    super();
  }

  get profil(): string {
    return this._profil;
  }

  set profil(value: string) {
    this._profil = value;
  }

  get ordinateur(): Ordinateur {
    return this._ordinateur;
  }

  set ordinateur(value: Ordinateur) {
    this._ordinateur = value;
  }

  get cursus(): Cursus {
    return this._cursus;
  }

  set cursus(value: Cursus) {
    this._cursus = value;
  }
}

File: java/src/main/java/com/sopra/Factory/model/Stagiaire.java
package com.sopra.Factory.model;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonView;
import com.sopra.Factory.model.view.JsonViews;

@Entity
@DiscriminatorValue("Stagiaire")
public class Stagiaire extends RessourceHumaine {

	@Enumerated(EnumType.STRING)
	@Column(name="profil")
	@JsonView(JsonViews.Common.class)
	private Profil profil;
	
	@ManyToOne
	@JoinColumn(name = "id_ressource_materielle")
	@JsonView(JsonViews.Common.class)
	private Ordinateur ordinateur;
	
	@ManyToOne
	@JoinColumn(name="id_cursus")
	@JsonView(JsonViews.Common.class)
	private Cursus cursus;

	public Stagiaire() {
		super();
	}

	public Profil getProfil() {
		return profil;
	}

	public void setProfil(Profil profil) {
		this.profil = profil;
	}

	public Ordinateur getOrdinateur() {
		return ordinateur;
	}

	public void setOrdinateur(Ordinateur ordinateur) {
		this.ordinateur = ordinateur;
	}

	public Cursus getCursus() {
		return cursus;
	}

	public void setCursus(Cursus cursus) {
		this.cursus = cursus;
	}

	
}
